database/factory/factory.go

```diff
@@ -6,59 +6,43 @@ package factory
 import (
 	"fmt"
 
-	"github.com/ava-labs/avalanchego/api/metrics"
 	"github.com/ava-labs/avalanchego/database"
 	"github.com/ava-labs/avalanchego/database/corruptabledb"
 	"github.com/ava-labs/avalanchego/database/leveldb"
 	"github.com/ava-labs/avalanchego/database/memdb"
-	"github.com/ava-labs/avalanchego/database/meterdb"
 	"github.com/ava-labs/avalanchego/database/pebbledb"
 	"github.com/ava-labs/avalanchego/database/versiondb"
 	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // New creates a new database instance based on the provided configuration.
 //
-// It also wraps the database with a corruptable DB and a meter DB.
+// It also wraps the database with a corruptable DB.
 //
 // dbName is the name of the database, either leveldb, memdb, or pebbledb.
 // dbPath is the path to the database folder.
 // readOnly indicates if the database should be read-only.
 // dbConfig is the database configuration in JSON format.
-// dbMetricsPrefix is used to create a new metrics registerer for the database.
-// meterDBRegName is used to create a new metrics registerer for the meter DB.
 func New(
```

Comment:
This function is used in subnet-evm, but the expected metrics by subnet-evm presumably differ from avalanchego. So I figured it would make the most sense to have subnet-evm register their own metrics as they wish.